### ISSUE TITLE
Multilabel Sequence Classification in trainer

### ISSUE THREAD

**Issue by GuillemGSubies** [huggingface/transformers#10232]
# 🚀 Feature request

We need to be able to use the trainer for multilabel classification problems.

## Motivation

Right now we create our models in the old fashioned way, with a sigmoid layer at the end so we can do multilabel. However if we could use the trainer directly, we wouldn't need to maintain different training scripts. Are there any plans for adding this to the trainer?

## Your contribution

I could try to help but I don't even know where to start with.

Thanks you very much for reading

**Comment by vimarshc** [comment]
Hi @LysandreJik, 
If no one is working on this can I start on this feature? 
I imagine this will not be that difficult and should be possible by using the sigmoid instead of the softmax, where we're calculating a probability between 0-1 for each class.

**Comment by LysandreJik** [comment]
Hi @vimarshc, I believe @abhishekkrthakur is working on this in https://github.com/huggingface/transformers/pull/11012

**Comment by vimarshc** [comment]
Thanks for the update! 
Shall try to make myself useful for some other issue. Haha.

**Comment by GuillemGSubies** [comment]
#11012 is merged and 4.6.0 is released, is this feature already there?

**Comment by LysandreJik** [comment]
I believe there is now multi-label classification within the models, and changing a model configuration argument (`config.problem_type = "multi_label_classification"`) should enable that out of the box. Have you tried it out?

**Comment by GuillemGSubies** [comment]
I did but I didn't have enough time to try it in depth, I had some problems with the labels format. However I did not find any documentation or examples in the docs.

**Comment by GuillemGSubies** [comment]
This seems to work but I found a weird problem that might be interesting to solve. 

When you binarize the labels, if you are training with pytorch it will throw an error because `BCEWithLogitsLoss` expects floats and not ints. This is counterintuitive for me, I had to cast my labels to floats and then it worked.

Also, there is not a lot of information about the anything of this in the documentation and the notebook for multi-label sequence classification still uses the old training loops, instead of the trainer.

**Comment by LysandreJik** [comment]
Indeed, it would be nice to put this in the documentation, it's sparse on this subject right now.

**Comment by GuillemGSubies** [comment]
Sorry for the flood but, is there a  way to instance a multilabel model on a pipeline? I'm trying it but I don't think it is working, the prediction probabilities always sum up to 1.

**Comment by LysandreJik** [comment]
I believe this is being taken care of in https://github.com/huggingface/transformers/pull/8328

**Comment by GuillemGSubies** [comment]
Indeed, it looks really good. Thank you

**Comment by xegulon** [comment]
Is the multilabel pipeline implemented in the last release `4.9.2` @LysandreJik ? If yes, are there examples on how to use it?

**Comment by LysandreJik** [comment]
Ssee the docs for the text classification pipeline [here](https://huggingface.co/transformers/master/main_classes/pipelines.html#transformers.TextClassificationPipeline) (See `function_to_apply` argument)

It is currently on the `master` branch and will be in the next release (v4.10.0)

**Comment by xegulon** [comment]
Great thanks!
However, I don't see a documentation which is specific to multilabel classification (I understand that choosing `function_to_apply='sigmoid'` makes it work though). With the results of the `pipeline` be prettified accordingly?

**Comment by gg22mm** [comment]
images/old/type1/tongue7226.jpg	0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0
images/old/type1/tongue165.jpg	0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0
images/old/type1/tongue11047.jpg	0,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0
images/old/type1/tongue1910.jpg	0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0
images/old/type10/tongue13658.jpg	1,0,0,1,1,0,0,1,0,1,0,1,0,1,0,0,0,1,0,1,0,0,0,1,0,0,0,0,0,1,0,0,1,0,0,1,1,0,1,1,0,1,0,1,0,1,0,1,0,1,0,0,0,1,0,0,0,0,0,0
images/old/type10/tongue12285.jpg	1,0,0,1,1,0,0,1,0,1,0,0,1,1,0,0,0,1,0,0,1,1,0,0,0,0,0,0,0,1,0,0,0,1,0,1,1,0,1,1,0,1,0,1,0,1,0,1,0,0,0,0,0,0,0,1,0,1,0,0
images/old/type1/tongue7157.jpg	0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0
images/old/type1/tongue3387.jpg	0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0

------------------------------------  batch=2  #torch.Size([2, 60]) ------------------------------------
I just want to know if the format of labels is correct? / 我只想知道labels是这样的格式是不是对的？ 

 'labels': tensor([ 
	 [0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.,
	 0., 0., 1., 0., 0., 0., 0., 0., 0., 1., 0., 1., 0., 0., 0., 0., 0., 0.,
	 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0.,
	 0., 0., 0., 0., 0., 0.],
	[0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.,
	 0., 0., 1., 0., 0., 0., 0., 0., 0., 1., 0., 1., 0., 0., 0., 0., 0., 0.,
	 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0.,
	 0., 0., 0., 0., 0., 0.]
])

**Comment by jbbqqf** [comment]
Hi! Triaging older issues this week — I think this one has shipped and can be closed.

Multilabel sequence classification with `Trainer` works today via `config.problem_type="multi_label_classification"`, which switches the classification heads to `BCEWithLogitsLoss`:

- `PretrainedConfig.problem_type` accepts `"multi_label_classification"`: [`src/transformers/configuration_utils.py#L250`](https://github.com/huggingface/transformers/blob/main/src/transformers/configuration_utils.py#L250).
- Docstring details: [`src/transformers/configuration_utils.py#L206`](https://github.com/huggingface/transformers/blob/main/src/transformers/configuration_utils.py#L206).

Typical usage: `AutoModelForSequenceClassification.from_pretrained(..., num_labels=N, problem_type="multi_label_classification")`, then feed float multi-hot labels to `Trainer`. No custom Trainer changes needed.

Would you mind closing this out? If a specific head or task variant is missing, happy to look.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually against the current `main` branch.)
